src/hooks/useFetch.ts

```diff
@@ -0,0 +1,25 @@
+import useSWR from "swr";
+import { SingleParam, addParams, fetcher } from "./utils";
+
+type Props = { requestParams: SingleParam[] };
+
+export const useFetch = ({ requestParams }: Props) => {
+  const apiKey = import.meta.env.VITE_API_KEY;
+  const endpoint = "http://www.omdbapi.com/";
+
+  const searchParams = addParams({
+    paramsArray: requestParams,
+    endpoint,
+  });
+
+  const { data, isLoading, error } = useSWR(
+    `${endpoint}?apikey=${apiKey}${searchParams}`,
+    fetcher,
+  );
+
+  return {
+    films: data?.Search,
+    isLoading,
+    error,
+  };
+};
```

Comment:
We briefly discussed it on our previous call - I suggest you to create dedicated api layer, that will only do requests:

```ts
// src/api.ts

type RequestOptions = {
    method: 'GET' | 'POST',
    queryParams?: Record<string, string>,
    // ...other options if needed
}

const API_BASE_URL = "http://www.omdbapi.com";

// Internal function to handle api key, response parsing, etc
const makeRequest = async (url: string, options?: RequestOptions) => {
    const queryParams = {
        apiKey: import.meta.env.VITE_API_KEY,
        ...options?.queryParams,
    }

    const url = `${API_BASE_URL}?${new URLSearchParams(queryParams)}`;

    const response = await fetch(url, {
        method: options?.method || 'GET',
        // ...other options if needed, e.g. 
    });

    if (!response.ok) {
        throw new Error('Failed to fetch data');
    }

    return await response.json();
}

export const fetchFilms = (search?: string) => makeRequest('/', { queryParams: { s: search } });
```

In that way you can use your api out of swr context (e.g. you might consider using react-router's loaders, etc)

Then you can use this api + useSwr hook directly in your component:

```ts
import { fetchFilms } from './api';

const { data, isLoading } = useSWR('films', fetchFilms);
```

And if you want to provide arguments, I recommend to not use `useSWR`'s key propagation, but rather pass arguments directly to the fetch function:

```ts
import { fetchFilms } from './api';

const [search, setSearch] = useState('');
const { data, isLoading } = useSWR(['films', search], () => fetchFilms(search));
```

This way you don't couple your api layer with swr, and can easily switch to another library if needed. Also, you can easily test your api layer without mocking swr's internals.